### ISSUE TITLE
Bug report:About train_acc and val_acc show err in visdom

### ISSUE THREAD

**Issue by zygmcc** [JinghuiZhou/awesome_face_antispoofing#12]
About main.py:line 229
In this part:
`vis.plot_many_stack({'train_loss': train_loss.value()[0], \
                             'val_loss': val_loss.value()[0]}, win_name="Loss")

 vis.plot_many_stack({'train_acc': train_acc.value()[0], \
                             'val_acc': val_acc.value()[0]}, win_name='Acc')`

When using GPU training，var train_acc and val_acc should use 
`.cpu()` to load in memory，So I change source code like this:
`        if opt.use_gpu:
            val_acc.add(v_accuracy.cpu())
        else:
            val_acc.add(v_accuracy)`
And
`        if opt.use_gpu:
            train_acc.add(epoch_acc.cpu())
        else:
            train_acc.add(epoch_acc)`

**Comment by zygmcc** [comment]
![image](https://user-images.githubusercontent.com/20073054/50599516-4c319f00-0ee9-11e9-88cb-cc5d72df635e.png)
![image](https://user-images.githubusercontent.com/20073054/50599518-4fc52600-0ee9-11e9-8f69-8bbddbd72bde.png)
On resnet18
I fell sad

**Comment by JaryHuang** [comment]
Hello,I think it is not err in visdom. First, I repeat training the resnet18 with batch_size=16 three times.I found:
 
I have Due to the train data is small. And the train data shuffle.  

> ![image](https://user-images.githubusercontent.com/20073054/50599516-4c319f00-0ee9-11e9-88cb-cc5d72df635e.png)
> ![image](https://user-images.githubusercontent.com/20073054/50599518-4fc52600-0ee9-11e9-8f69-8bbddbd72bde.png)
> On resnet18
> I fell sad

![image](https://user-images.githubusercontent.com/35137503/51070740-43776080-1681-11e9-99a1-bd7ad19866cf.png)


> ![image](https://user-images.githubusercontent.com/20073054/50599516-4c319f00-0ee9-11e9-88cb-cc5d72df635e.png)
> ![image](https://user-images.githubusercontent.com/20073054/50599518-4fc52600-0ee9-11e9-8f69-8bbddbd72bde.png)
> On resnet18
> I fell sad

Hello,I think it is not err in visdom. First, I repeat training the resnet18 with batch_size=16 three times.I found:
![screenshot from 2019-01-12 15-45-19](https://user-images.githubusercontent.com/35137503/51070776-b680d700-1681-11e9-9a8f-cf7af3517f79.png)
So,I guee the reason may be as follows:
1.Due to the train data is small. And the train data shuffle.
2.the resnet18 is not robust enough.
I change the batch_size=64
![screenshot from 2019-01-12 15-53-16](https://user-images.githubusercontent.com/35137503/51070804-40c93b00-1682-11e9-971b-91202b566a8a.png)

**Comment by JaryHuang** [comment]
> ![image](https://user-images.githubusercontent.com/20073054/50599516-4c319f00-0ee9-11e9-88cb-cc5d72df635e.png)
> ![image](https://user-images.githubusercontent.com/20073054/50599518-4fc52600-0ee9-11e9-8f69-8bbddbd72bde.png)
> On resnet18
> I fell sad
Hello, I think it is not err in visdom. First,I repeat training the resnet18 with batch_size=16 three times.I found:
![screenshot from 2019-01-12 15-45-19](https://user-images.githubusercontent.com/35137503/51070862-280d5500-1683-11e9-857e-cecdb50312c2.png)
it is  unstable.
So,I guee the reason may be as follows:
1.Due to the train data is small. And the train data shuffle.
2.the resnet18 is not robust enough.
I change the batch_size=64
![screenshot from 2019-01-12 15-53-16](https://user-images.githubusercontent.com/35137503/51070867-2fccf980-1683-11e9-81ec-bc2963f83abc.png)
it is relatiuvely stable. so you bug just is the training error.

**Comment by JaryHuang** [comment]
> ![image](https://user-images.githubusercontent.com/20073054/50599516-4c319f00-0ee9-11e9-88cb-cc5d72df635e.png)
> ![image](https://user-images.githubusercontent.com/20073054/50599518-4fc52600-0ee9-11e9-8f69-8bbddbd72bde.png)
> On resnet18
> I fell sad
Hello, I think it is not err in visdom. First,I repeat training the resnet18 with batch_size=16 three times.I found:
![screenshot from 2019-01-12 15-45-19](https://user-images.githubusercontent.com/35137503/51070910-06f93400-1684-11e9-92ad-0ae93d44cf2c.png)
it is unstable.
So,I guee the reason may be as follows:
1.Due to the train data is small. And the train data shuffle.when batch=16,is it small
2.the resnet18 is not robust enough.
I just change the batch_size=64
![screenshot from 2019-01-12 15-53-16](https://user-images.githubusercontent.com/35137503/51070917-2728f300-1684-11e9-8ffa-7e6f9af9e7f2.png)
it is relatiuvely stable. so you bug just is the training error.

**Comment by justein** [comment]
@zygmcc would you mind share the way to solve the 'tensor(acc=0.985,dtype=...) is not JSON serializable' problem ?
